manifests/server.pp

```diff
@@ -158,18 +159,43 @@
           }
         }
       }
+    }
 
-      # Wrap the replset class
-      class { 'mongodb::replset':
-        sets => $replset_config_REAL
-      }
-      Anchor['mongodb::server::end'] -> Class['mongodb::replset']
+    # Wrap the replset class
+    class { 'mongodb::replset':
+      sets => $replset_config_REAL
+    }
+    Anchor['mongodb::server::end'] -> Class['mongodb::replset']
+
+    # Make sure that the ordering is correct
+    if $create_admin {
+      Class['mongodb::replset'] -> Mongodb::Db['admin']
+      if $::mongodb_is_master == 'not_installed' and $auth == true and $noauth != true and versioncmp($version, '2.6.0') >= 0  {
+        file_line{ 'enable_authentication' :
+          ensure  =>  present,
+          path    => $config,
+          match   => 'security.authorization:',
+          line    => 'security.authorization: enabled',
+          require => [Class['mongodb::replset'], Mongodb::Db['admin'] ]
+        }
+        if $keyfile {
+          file_line{ 'enable_keyfile' :
+            ensure  =>  present,
+            path    => $config,
+            line    => "security.keyFile: ${keyfile}",
+            require => [Class['mongodb::replset'], Mongodb::Db['admin']],
+            notify  => Exec['/sbin/restart mongod']
+          }
+        }
 
-      # Make sure that the ordering is correct
-      if $create_admin {
-        Class['mongodb::replset'] -> Mongodb::Db['admin']
+        exec{ '/sbin/restart mongod':
```

Comment:
Could you explain why this is here instead of using a service resource or migrating the exec elsewhere?